File: GO/zigzag_conversion.go
// Link: https://leetcode.com/problems/zigzag-conversion/
func convert(s string, numRows int) string {
    s_size := len(s)
	ans := ""
	if numRows == 1 {
		return s
	}
	cycleLen := (numRows - 1) * 2;
	for i := 0; i < numRows; i++ {
		for j := 0; j < s_size-i; j = j + cycleLen {
			ans += string(s[j+i])
			if i != 0 && i != numRows-1 && j < s_size+i-cycleLen {
				ans += string(s[j-i+cycleLen])
			}
		}
	}
	return ans
}

File: Python/ZigZag_Conversion.py
# Problem Link: https://leetcode.com/problems/zigzag-conversion/

class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if (numRows <= 1):
            return s

        elif (numRows == 2):
            output1 = ""
            output2 = ""
            for i in range(len(s)):
                if (i % 2 == 0):
                    output1 += s[i]
                else:
                    output2 += s[i]
            return (output1 + output2)

        else:
            output = []
            direction = -1
            rowNumber = 0

            for i in range(numRows):
                output.append("")

            for j in range(len(s)):
                output[rowNumber] += s[j]
                if ((rowNumber == 0) or (rowNumber == numRows - 1)):
                    direction *= -1
                rowNumber += direction

            return ("".join(output))